util/src/main/scala/scala/scalanative/io/ByteBufferPool.scala

```diff
@@ -7,7 +7,7 @@ final class ByteBufferPool {
   private var buffers: List[ByteBuffer] = Nil
 
   private def alloc(): ByteBuffer = {
-    ByteBuffer.allocateDirect(8 * 1024 * 1024)
+    ByteBuffer.allocateDirect(32 * 1024 * 1024)
```

Comment:
@EmergentOrder   Thank you taking the time to create this PR.

I think you have identified an area which needs a bit of re-design & re-work,
possibly outside the scope of what you signed on to do.

Changing from one fixed constant to another sets up the scenario for dueling
constants no matter what the constant.   

I scratched around a bit and the Feb 18, 2017 commit #61ce63b57b65e8e99bd5e873030c3e911e988829
specifically decreases the fixed size from (128 * 1024 * 1024) and notes that change.  Now there 
may be a sweet spot larger than the current 8 and less than 128, but probably not, 

Are you getting to ByteBufferPool by using the io package object method withScratchBuffer() or
by creating a ByteBufferPool directly?

If the former, withScratchBuffer is used by NirGenFile.scala & CodeGen.scala.  This PR may also 
affect Nir & Code Generation.  @densh is the expert there.

If you are creating the ByteBufferPool yourself, i suggest that you consider a solution where the class has
a second constructor, with two arguments.  One would be the size of each fixed size buffer. The second
would be the number of fixed size buffers to retain on reclaim.  

From my reading, these direct buffers are actually being allocated on the heap.  Discussions of more
adventurous  solutions should probably be done by creating an Issue, rather than here is a PR.

Reply:
@LeeTibbert Thanks for the feedback. I'm not dealing with the ByteBufferPool in either of the ways you mentioned. I'm just trying to build my project with Scala Native using sbt. I'm aware there implications to this change, and the value of 32 here won't be optimal. But it does unblock my and others' usage of Scala Native while remaining within the bounds of the previous values and passing CI.

Reply:
@EmergentOrder,

The information that you are not directly doing anything with ByteBufferPool is quite useful
Thank you for taking the time to make SN better for all.

ByteBufferPool is used by the SN backend, so you may have identified an issue 
with the backend handling either large files or/and large projects. 

BTW, Whilst chasing this I took a look at your project and it looks _amazing_.

Reply:
@LeeTibbert 
I was thinking the same thing. 
In my case, compiling this large (~3k lines) file triggers the exception: [ONNX.scala](https://github.com/EmergentOrder/onnx-scala/blob/master/core/src/main/scala-2.11/ONNX.scala)

Thanks for the kind words re: my project, and also for all your work on SN.